### PR TITLE
Cleanup/login

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/provokateurin/nextcloud-neon/issues/470
Addresses problems found in https://github.com/provokateurin/nextcloud-neon/pull/291

### CODE REVIEW

File: packages/neon/neon/lib/src/pages/login.dart [hunk 1]
```diff
@@ -56,48 +56,27 @@ class _LoginPageState extends State<LoginPage> {
...
                   ],
+                  const SizedBox(
+                    height: 60,
+                  ),
                   Form(
                     key: _formKey,
                     child: TextFormField(
                       focusNode: _focusNode,
-                      decoration: const InputDecoration(
+                      decoration: InputDecoration(
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/pages/login.dart:76]: We should also provide a submit button as not everyone knows the submit functionality of the keyboard (a11y)

**Review by Leptopoda — APPROVED** [review]
The UX is better but I still don't like it; I imagined something more like:

[patch.txt](https://github.com/provokateurin/nextcloud-neon/files/12063549/patch.txt)

**Comment by Leptopoda** [comment]
https://github.com/provokateurin/nextcloud-neon/assets/25266387/a165f997-b8a7-4c1d-9890-34bf3bcb49fa

**Comment by provokateurin** [comment]
@Leptopoda I applied your suggestions but didn't apply the patch. Using an animation is a good idea but out of scope for this PR imo. I saw you tried adding better error handling, but I'm a bit confused by that because we already use NeonException which will display the most accurate error message possible :thinking:

File: packages/neon/neon/lib/src/widgets/nextcloud_logo.dart [hunk 2]
```diff
@@ -12,5 +13,6 @@ class NextcloudLogo extends StatelessWidget {
         package: 'neon',
         width: 100,
         height: 100,
+        semanticsLabel: AppLocalizations.of(context).nextcloudLogo,
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/widgets/nextcloud_logo.dart:16]: I think the normal nextcloud brand name `Nextcloud` would be enough.

My Idea was to have something like "Neon \n works with \n Nextcloud" while we now get "Works with \n Nextcloud logo".

(The neon logo should also get a semantic node in a followup)

**Comment by Leptopoda** [comment]
>  I saw you tried adding better error handling, but I'm a bit confused by that because we already use NeonException which will display the most accurate error message possible thinking

I only did this for design reasons and not for the text we show. The above video shows how I'd like the design (my opinion and I'm not a designer by any means).

- If an error occurs the checks should be grayed out to show they weren't run/completed.
- the retry button should replace the continue one: A user already expects to press this button and not a separate one to pop up somewhere else
- When the failed checks are still shown (why shouldn't they) the "NeonException" should follow the design of the other checks and not be centered.

Does this make sense?

File: packages/neon/neon/lib/src/widgets/nextcloud_logo.dart (hunk 2 above)
Comment (provokateurin) [reply packages/neon/neon/lib/src/widgets/nextcloud_logo.dart:16]: No the logo on the login page is wrapped with extra semantics exactly because of this.

**Comment by provokateurin** [comment]
Now I understand it better. The patch was a bit hard to understand. I'll try to implement your ideas :+1:

**Comment by provokateurin** [comment]
I haven't implemented the size transition, but everything else.

**Comment by Leptopoda** [comment]
I think we could move the login submit button into the server URL field as a suffix icon (similar to what the NC app does).

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**